### PR TITLE
Add warnings when provider unbonding is shorter than consumer unbonding

### PR DESCRIPTION
**Pull request by p-offtermatt**
# Description

Changes the client code for creating consumer addition proposals so that a warning is printed when
the unbonding period in the proposal is >= than the unbonding period of the provider.

To test anything here, I needed to create a mock of the CometBFT RPC client. I didn't find anything like this elsewhere, but lmk if it already exists.

For the error message, I don't think the logger in the app is in scope here, so I just print to stderr.

## Linked issues

Closes: #733 

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [x] `Feature`: Changes and/or adds code behavior, irrelevant to bug fixes
- [ ] `Fix`: Changes and/or adds code behavior, specifically to fix a bug
- [ ] `Refactor`: Changes existing code style, naming, structure, etc.
- [x] `Testing`: Adds testing
- [ ] `Docs`: Adds documentation

## New behavior tests

Added a test for the logic that checks unbonding periods of provider and consumer against each other.

## Versioning Implications

- [ ] This PR will affect [semantic versioning as defined for ICS](../CONTRIBUTING.md#semantic-versioning)

If the above box is checked, which version should be bumped?

- [ ] `MAJOR`: Consensus breaking changes to both the provider and consumers(s), including updates/breaking changes to IBC communication between provider and consumer(s)
- [ ] `MINOR`: Consensus breaking changes which affect either only the provider or only the consumer(s)
- [ ] `PATCH`: Non consensus breaking changes

## Targeting

Please select one of the following:

- [x] This PR is only relevant to main
- [ ] This PR is relevant to main, and should also be back-ported to ____ (ex: v1.0.0 and v1.1.0)
- [ ] This PR is only relevant to ____ (ex: v1.0.0, v1.1.0, and v1.2.0)

### CODE REVIEW

**Comment by p-offtermatt** [comment]
The quality gate is a bit unhelpful here: The new code is in an area where we have 0% test coverage at the moment, since it is concerned with the CLI, for which there are no unit tests. I added tests for my new functionality as much as possible. I propose ignoring the code coverage warning for this one.

File: testutil/mock_rpc_client.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	"context"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/tendermint/tendermint/libs/bytes"
+	"github.com/tendermint/tendermint/libs/log"
+	rpcClient "github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+)
+
```
[the remark this answers is not in the document]
Comment (p-offtermatt) [reply testutil/mock_rpc_client.go:13]: Great point. This was generated with the help of an LLM. I had trouble getting the mockgen to work for this, but you are right, this should be consistent. I will take a look

File: x/ccv/provider/client/proposal_handler.go [hunk 2]
```diff
@@ -74,6 +75,16 @@ Where proposal.json contains:
 				return err
 			}
 
+			// do not fail for errors regarding the unbonding period, but just log a warning
+			err = CheckPropUnbondingPeriod(clientCtx, proposal.UnbondingPeriod)
+			if err != nil {
+				fmt.Fprintf(
```
Comment (MSalopek) [inline x/ccv/provider/client/proposal_handler.go:81]: Hey, I'm bikeshedding but why not just do this?
```suggestion
fmt.Errorf("Warning: Could not assure that Proposal Unbonding Period is shorter than Provider Unbonding Period. Error message: %s, err.Error()
```

File: x/ccv/provider/client/proposal_handler.go [hunk 3]
```diff
@@ -441,3 +452,29 @@ func postEquivocationProposalHandlerFn(clientCtx client.Context) http.HandlerFun
...
+	if err != nil {
+		return err
+	}
+
+	consensusParams, err := client.ConsensusParams(context.Background(), nil)
+	if err != nil {
+		return err
+	}
+
+	providerUnbondingTime := consensusParams.ConsensusParams.Evidence.MaxAgeDuration
```
Comment (MSalopek) [inline x/ccv/provider/client/proposal_handler.go:467]: Not sure about this, maxAgeDuration is not the same as the unbonding period. The evidence age is configured separately from the unbonding period, those are 2 different parameters.

Comment (p-offtermatt) [reply x/ccv/provider/client/proposal_handler.go]: You are right, has been adjusted

File: x/ccv/provider/client/proposal_handler.go (hunk 2 above)
Comment (p-offtermatt) [reply x/ccv/provider/client/proposal_handler.go:82]: Not clear what you mean by this. What you wrote creates an error object, but I want to primt to Stderr.

File: x/ccv/provider/client/proposal_handler.go [hunk 4]
```diff
@@ -441,3 +452,29 @@ func postEquivocationProposalHandlerFn(clientCtx client.Context) http.HandlerFun
...
+
+	if providerUnbondingTime <= propUnbondingPeriod {
+		return fmt.Errorf(
+			"consumer unbonding period should be smaller than provider unbonding period, but is longer: \n"+
```
[the remark this answers is not in the document]
Comment (p-offtermatt) [reply x/ccv/provider/client/proposal_handler.go:469]: Clarified the error message, thanks

File: x/ccv/provider/client/proposal_handler.go [hunk 5]
```diff
@@ -441,3 +452,29 @@ func postEquivocationProposalHandlerFn(clientCtx client.Context) http.HandlerFun
...
+				"consumer unbonding: %s \n provider unbonding: %s \n"+
+				"See the Consumer onboarding documentation for more information:\n"+
+				"https://cosmos.github.io/interchain-security/consumer-development/onboarding#3-submit-a-governance-proposal",
```
[the remark this answers is not in the document]
Comment (p-offtermatt) [reply x/ccv/provider/client/proposal_handler.go:472]: Thanks, I agree

File: testutil/mock_rpc_client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (p-offtermatt) [reply testutil/mock_rpc_client.go]: Removed this, since the new method relies on a different interface

**Review by MSalopek — APPROVED** [review]
New changes look good.

Approve!

**Review by smarshall-spitzbart — APPROVED** [review]
One comment but LGTM!

File: x/ccv/provider/client/proposal_handler.go [hunk 6]
```diff
@@ -441,3 +452,26 @@ func postEquivocationProposalHandlerFn(clientCtx client.Context) http.HandlerFun
...
+	queryClient := stakingtypes.NewQueryClient(clientCtx)
+
+	res, err := queryClient.Params(context.Background(), &stakingtypes.QueryParamsRequest{})
+	if err != nil {
+		return err
+	}
+
+	providerUnbondingTime := res.Params.UnbondingTime
+
+	if providerUnbondingTime < propUnbondingPeriod {
```
[the remark this answers is not in the document]
Comment (p-offtermatt) [reply x/ccv/provider/client/proposal_handler.go:461]: Thanks, good suggestion imo

**Pull request merged**